webpack/assets/javascripts/react_app/components/UserJwtToken/index.js

```diff
@@ -0,0 +1,69 @@
+import React, { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import { get } from '../../redux/API';
+import { USER_JWT_TOKEN } from './UserJwtTokenConstants';
+import UserJwtTokenForm from './components/UserJwtTokenForm';
+import TokenField from './components/TokenField';
+import { selectStatus, selectUserJwtToken } from './UserJwtTokenSelectors';
+
+const UserJwtToken = ({ userId }) => {
+  const dispatch = useDispatch();
+  const apiStatus = useSelector(selectStatus);
+  const token = useSelector(selectUserJwtToken);
+
+  const [unit, setUnit] = useState('never');
+  const [expValue, setExpValue] = useState('');
+
+  const changeUnit = u => {
+    if (u === 'never') {
+      setExpValue('');
+    }
+    setUnit(u);
+  };
+
+  const handleSubmit = () => {
+    const params = {
+      exp_value: expValue,
+      exp_unit: unit,
+    };
+
+    dispatch(
+      get({
+        key: USER_JWT_TOKEN,
+        url: `/users/${userId}/jwt_token`,
+        params,
+      })
+    );
+  };
+
+  const copyToken = () => {
+    const tmpElement = document.createElement('textarea');
+
+    tmpElement.appendChild(document.createTextNode(token));
+    document.body.appendChild(tmpElement);
+    tmpElement.select();
+    document.execCommand('copy');
+    document.body.removeChild(tmpElement);
+  };
+
+  return (
+    <div>
+      <UserJwtTokenForm
+        expValue={expValue}
+        setExpValue={setExpValue}
+        expUnit={unit}
+        setExpUnit={changeUnit}
```

Comment:
maybe all of this state handlers should be moved into `UserJwtTokenForm` component
I don't see other usage for them